File: javasCE2/ClasesVO/ListaVO.java
package javasCE2.ClasesVO;
import java.io.Serializable;

public class ListaVO implements Serializable {
    private static final long serialVersionUID = 1L;

    private String listaId;

    public ListaVO(){}

    public ListaVO(String listaId) {
        this.listaId = listaId;
    }

    public String getListaId() {
        return listaId;
    }

    public void setListaId(String listaId) {
        this.listaId = listaId;
    }

    @Override
    public String toString() {
        return "ListaVO{" +
                "ListaId='" + listaId + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ListaVO listaVO = (ListaVO) o;
        return listaId != null && listaId.equals(listaVO.listaId);
    }

    @Override
    public int hashCode() {
        return listaId != null ? listaId.hashCode() : 0;
    }
}

File: VO php/ListaVO.php
<?php
class ListaVO {
    
    private $listaId;

    // Constructor
    public function __construct($listaId = null) {
        $this->listaId = $listaId;
    }

    // Getter
    public function getListaId() {
        return $this->listaId;
    }

    // Setter
    public function setListaId($listaId) {
        $this->listaId = $listaId;
    }

    // toString -->  ESTO DEBERÍA IR EN LA CLASES VAO
    public function toString() {
        return "ListaVO{" .
                "ListaId='" . $this->listaId . "'" .
                '}';
    }

    // Método para comprobar igualdad basado en listaId -->  ESTO DEBERÍA IR EN LA CLASES VAO
    public function equals($obj) {
        if ($this === $obj) return true;
        if ($obj === null || get_class($this) !== get_class($obj)) return false;
        return $this->listaId !== null && $this->listaId === $obj->getListaId();
    }
}
?>